src/scheduled_tasks.rs

```diff
@@ -404,9 +431,8 @@ fn update_banned_when_expired(conn: &mut PgConnection) {
 
 /// Updates the instance software and version
 ///
-/// TODO: this should be async
 /// TODO: if instance has been dead for a long time, it should be checked less frequently
-fn update_instance_software(conn: &mut PgConnection, user_agent: &str) -> LemmyResult<()> {
+async fn update_instance_software(conn: &mut PgConnection, user_agent: &str) -> LemmyResult<()> {
   info!("Updating instances software and versions...");
 
   let client = Client::builder()
```

Comment:
Here you should be able to pass in context and use the existing client.

Reply:
Good point, I'll do that